### PR TITLE
samples: net: MQTT: Use Zephyr NET and Connection mgmr for nRF91 boards

### PR DESCRIPTION
**Pull request by simensrostad**
Combine `network_wifi.c` and `network_lte.c` modules into `network.c`. Both network stacks are now managable via Zephyr NET and Connection manager APIs.

### CODE REVIEW

File: samples/nrf9160/aws_iot/README.rst
Comment (jhn-nordic) [inline samples/nrf9160/aws_iot/README.rst:1]: should not be in this PR?

**Review by divipillai — APPROVED** [review]

File: samples/net/mqtt/src/modules/network/network.c [hunk 1]
```diff
@@ -64,30 +67,35 @@ static void connectivity_event_handler(struct net_mgmt_event_callback *cb,
...
 	}
 
-	LOG_INF("Connecting...");
-
-	err = conn_mgr_if_connect(net_if);
-	if (err) {
-		LOG_ERR("conn_mgr_if_connect, error: %d", err);
-		SEND_FATAL_ERROR();
-		return;
+	/* Resend connection status if the sample is build for Native Posix.
```
Comment (jhn-nordic) [inline samples/net/mqtt/src/modules/network/network.c:91]: built for

**Review by jhn-nordic — APPROVED** [review]

**Comment by maxd-nordic** [comment]
Could you rename the overlays as well? There are also more occurrences of nrf9160 in the CMakeLists and Kconfig.
Similar with nrf7002.

**Comment by simensrostad** [comment]
> Could you rename the overlays as well? There are also more occurrences of nrf9160 in the CMakeLists and Kconfig. Similar with nrf7002.

done

**Review by maxd-nordic — APPROVED** [review]
lgtm

**Review by rlubos — APPROVED** [review]

**Pull request merged**